File: src/main/java/org/demo/domain/OverdraftDetectedException.java
package org.demo.domain;

import java.math.BigDecimal;

public class OverdraftDetectedException extends Exception {
	public OverdraftDetectedException(String accountNumber, BigDecimal balance, BigDecimal withdrawalAmount) {
		super(String.format("Overdraft detected on account %s [current balance: %.2f, requested amount: %.2f]",
				accountNumber, balance, withdrawalAmount));
	}
}

File: src/main/java/org/demo/domain/OverdraftDetectedException.kt
package org.demo.domain

import java.math.BigDecimal

class OverdraftDetectedException(accountNumber: String?, balance: BigDecimal?, withdrawalAmount: BigDecimal?) : Exception(String.format("Overdraft detected on account %s [current balance: %.2f, requested amount: %.2f]",
        accountNumber, balance, withdrawalAmount))
